ICSharpCode.CodeConverter/CSharp/CommonConversions.cs

```diff
@@ -51,7 +52,9 @@ public CommonConversions(SemanticModel semanticModel, VisualBasicSyntaxVisitor<C
                 bool isField = declarator.Parent.IsKind(SyntaxKind.FieldDeclaration);
                 EqualsValueClauseSyntax equalsValueClauseSyntax;
                 if (adjustedInitializer != null) {
-                    equalsValueClauseSyntax = SyntaxFactory.EqualsValueClause(adjustedInitializer);
+                    var vbInitializer = declarator.Initializer?.Value;
+                    var convertedInitializer = vbInitializer == null ? adjustedInitializer : TypeConversionAnalyzer.AddExplicitConversion(vbInitializer, adjustedInitializer);
```

Comment:
Explicit conversions are never needed for `AsClause`, since the type is inferred from the RHS. Probably should have left a comment about that here.